docs/usage/serve.rst

```diff
@@ -39,3 +39,32 @@ locations like ``/etc/environment`` or in the forced command itself (example bel
     has been introduced in v7.2. We recommend to use
     ``no-port-forwarding,no-X11-forwarding,no-pty,no-agent-forwarding,no-user-rc``
     in this case.
+    
+SSH Configuration
+~~~~~~~~~~~~~~~~~
+``borg serve``'s pipes (``stdin``/``stdout``/``stderr``) are connected to the ``sshd`` process on the server side. In the event that the SSH connection between ``borg serve`` and the client is disconnected abnormally (for example, due to a network outage), it can take a long time for ``sshd`` to notice the client is disconnected. In the meantime, ``sshd`` continues running, and as a result so does the ``borg serve`` process holding the lock on the repository. This can cause subsequent ``borg`` operations on the remote repository to fail with the error: ``Failed to create/acquire the lock``.
+
+In order to avoid this, it is recommended to perform the following additional SSH configuration:
+
+In the client side's ``~/.ssh/config`` file:
```

Comment:
on the server side, you do the config change globally, but here only for the backupserver host.

any special reason for this? it might count as defensive, but on the other hand: which scenario could there be that does not want to terminate the connection after it has been non-working for 5 minutes?

Reply:
This is deliberate. When I add the `ClientAliveInterval` and `ClientAliveCountMax` within a match block, eg:

```
Match User backupuser
    AllowTCPForwarding no
    X11Forwarding no
    ClientAliveInterval 10
    ClientAliveCountMax 30
```

...when sshd is reloaded, it terminates the the following errors in syslog:

```
Apr  5 10:47:10 frac systemd[1]: Reloading OpenBSD Secure Shell server.
Apr  5 10:47:10 frac systemd[1]: Reloaded OpenBSD Secure Shell server.
Apr  5 10:47:10 frac sshd[14143]: /etc/ssh/sshd_config line 98: Directive 'ClientAliveInterval' is not allowed within a Match block
Apr  5 10:47:10 frac systemd[1]: ssh.service: Main process exited, code=exited, status=255/n/a
Apr  5 10:47:10 frac systemd[1]: ssh.service: Unit entered failed state.
Apr  5 10:47:10 frac systemd[1]: ssh.service: Failed with result 'exit-code'.
```

Thus, it looks like these settings need to be made globally on the server side.

On the client side, there is no issues adding these to the user's `~/.ssh/config` file as-per what I've written.

Reply:
If you'd like, I can add some text letting users know the server side configuration items need to be global, to prevent them from trying it in a match block and breaking their sshd and locking themselves out of their server. :-)

Reply:
> it might count as defensive, but on the other hand: which scenario could there be that does not want to terminate the connection after it has been non-working for 5 minutes?

I've given this some thought, and I honestly can't think of any... Possibly on highly latent links (satellite) where there may be long periods of inactivity? Hence my "adjust to suit your environment" comment. :-)

Reply:
what i meant is "why not make it global in /etc/ssh/ssh_config?".

Reply:
Yep, we could do that. I'll adjust the wording.

Reply:
Wording updated in commit f9d011be